Exercise_2.java

```diff
@@ -9,37 +9,62 @@
         StackNode(int data) 
         { 
             //Constructor here 
+        	this.data = data;
         } 
     } 
     
 	
     public boolean isEmpty() 
     { 
         //Write your code here for the condition if stack is empty. 
+    	if(root ==  null) {
+    		System.out.println("Stack Empty");
+    		return true;
+    	}
+    	return false;
     } 
   
     public void push(int data) 
     { 
         //Write code to push data to the stack. 
+    	StackNode temp = new StackNode(data);
+    	if(temp == null) 
+    		System.out.println("Stack Overflow");
+    	
+    	temp.data = data;
+    	temp.next = root;
+    	root = temp;
     } 
   
     public int pop() 
     { 	
 	//If Stack Empty Return 0 and print "Stack Underflow"
+    	if(isEmpty()) {
+    		System.out.println("Stack Underflow");
+    		return 0;
+    	}
         //Write code to pop the topmost element of stack.
+    	root = root.next;
 	//Also return the popped element 
+		return root.data;

```

Comment:
You will have to store the root node before reassigning it, this way you are losing the pointer and returning the incorrect value.